### PR TITLE
generate function bugfix

### PR DESCRIPTION
**Pull request by thomasfischer11**
added keyword arguments & cut off generation if max_steps < config.seq_len

### CODE REVIEW

File: test/test_generate.jl [hunk 1]
```diff
@@ -49,16 +49,30 @@ using Test
...
+    @testset "cutoff at max_steps" begin
+        output = generate(
+            transformer,
+            tokenizer,
+            sampler_1,
+            "",
+            verbose=false, 
+            display_output=false, 
+            display_prompt=false,
+            max_steps=10,
```
Comment (janik072) [inline test/test_generate.jl:61]: Please stay consistent whether you use the optional comma after the last parameter or not.
Check every call of the generate function in the test file.

File: src/generate.jl [hunk 2]
```diff
@@ -15,14 +15,12 @@ function generate(
     model::Transformer{T},
     tokenizer::Tokenizer,
     sampler::Sampler{T},
-    prompt::String,
+    prompt::String;
```
Comment (janik072) [inline src/generate.jl:18]: Why did you choose to go with a different separator for this parameter? Not complaining about functionality, just curious about the design decision.

**Review by janik072 — COMMENTED** [review]
Looks fine, just noted some design inconsistencies.

File: src/generate.jl (hunk 2 above)
Comment (thomasfischer11) [reply src/generate.jl:18]: https://docs.julialang.org/en/v1/manual/functions/#Keyword-Arguments

**Review by kleincode — APPROVED** [review]

**Pull request merged**